eip_operator/src/main.rs

```diff
@@ -149,6 +159,30 @@ async fn run() -> Result<(), Error> {
         task::spawn(eip_controller.run())
     });
 
+    tasks.push({
+        let eip_api = eip_api.clone();
+        let node_api = node_api.clone();
+        let watch_config =
+            kube_runtime::watcher::Config::default().labels(EGRESS_GATEWAY_NODE_SELECTOR_LABEL_KEY);
+
+        task::spawn(async move {
+            let mut watcher = pin!(kube_runtime::watcher(node_api.clone(), watch_config));
```

Comment:
hm I'd think we'd want to watch `Eip` resources rather than `Node`, since we set the Eip's status each time the Eip is associated or disassociated. it's not clear to me that this fires if we associate an Eip right now

separately, this watcher is also self-triggering, where setting the egress gateway status will trigger another reconciliation. not a correctness issue if it converges, but could create some read/write amplification or hard-to-debug issues if it does anything unexpected

Reply:
Good find! The watcher is now correctly configured on Eip resources.